modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/TableRowConverterImpl.java

```diff
@@ -115,15 +80,7 @@ public class TableRowConverterImpl implements TableRowConverter {
     public <RowT> BinaryRowEx toBinaryRow(ExecutionContext<RowT> ectx, RowT row, boolean key) {
         BinaryTuple binaryTuple = ectx.rowHandler().toBinaryTuple(row);
 
-        if (!key) {
-            InternalTuple tuple = schemaDescriptor.physicalOrderMatchesLogical()
-                    ? binaryTuple
-                    : new FormatAwareProjectedTuple(binaryTuple, fullMapping);
-
-            return SqlOutputBinaryRow.newRow(schemaDescriptor.version(), fullRowColocationIndexes, colocationColumnTypes, tuple);
-        } else {
-            return SqlOutputBinaryRow.newRow(schemaDescriptor.version(), keyOnlyColocationIndexes, colocationColumnTypes, binaryTuple);
-        }
+        return SqlOutputBinaryRow.newRow(schemaDescriptor, key, binaryTuple);
```

Comment:
Is it possible to get rid of `boolean key` flag?
Having a special method usually better than using a flag, when the caller always knows the case.

It is ok to do in a separate ticket.

Reply:
make sense. I filed an issue https://issues.apache.org/jira/browse/IGNITE-21731